plugins/modules/zabbix_globalmacro.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2013-2014, Epic Games, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: zabbix_globalmacro
+short_description: Create/update/delete Zabbix Global macros
+version_added: 1.4.0
+description:
+   - manages Zabbix Global macros, it can create, update or delete them.
+   - For macro_type 1 (Secret Text) the value field cannot be validated and will always be overwritten due to the secret nature of the Text.
+author:
+    - "Cove (@cove)"
+    - Dean Hailin Song (!UNKNOWN)
+    - Timothy Test (@ttestscripting)
+requirements:
+    - "python >= 2.6"
+    - "zabbix-api >= 0.5.4"
+options:
+    macro_name:
+        description:
+            - Name of the global macro in zabbix native format C({$MACRO}) or simple format C(MACRO).
+        required: true
+        type: str
+    macro_value:
+        description:
+            - Value of the global macro.
+            - Required if I(state=present).
+        type: str
+    macro_type:
+        description:
+            - Type of the global macro Text or Secret Text.
+            - Required if I(state=present).
+            - 0 Text
+            - 1 Secret Text Works only with Zabbix >= 5.0 and will default to Text in lower versions
+            - 2 Vault secret Works only with Zabbix >= 5.2 and will default to Text in lower versions
+        type: str
+        choices: ['0', '1', '2']
```

Comment:
forgot one more thing, these should ideally be text values `['text', 'secret', 'vault']` or something similar and you need to do conversion to integers in the module. Its is sort of Ansible way of doing things that users should not consult API documentation when working with modules (although not doable 100% of a time)

Reply:
sure let me resolve that quickly, I will probably need to re-write the tests quickly as well